File: GanServer/src/com/xattacker/binary/BinaryReadable.java
package com.xattacker.binary;

import java.io.IOException;

public interface BinaryReadable
{
	long available();

	byte[] readBinary(int aLength) throws IOException; 
	short readShort() throws IOException;
	int readInteger() throws IOException;
	long readLong() throws IOException;
	double readDouble() throws IOException;
	String readString() throws IOException;
}

File: GanAndroidApp/GanClient/src/main/java/com/xattacker/binary/BinaryReadable.kt
package com.xattacker.binary

import java.io.IOException

interface BinaryReadable
{
    val available: Long

    @Throws(IOException::class)
    fun readBinary(aLength: Int): ByteArray?

    @Throws(IOException::class)
    fun readShort(): Short?

    @Throws(IOException::class)
    fun readInteger(): Int?

    @Throws(IOException::class)
    fun readLong(): Long?

    @Throws(IOException::class)
    fun readDouble(): Double?

    @Throws(IOException::class)
    fun readString(): String?
}
